lambdas/services/document_manifest_service.py

```diff
@@ -6,33 +6,56 @@
 from botocore.exceptions import ClientError
 from models.document_reference import DocumentReference
 from models.zip_trace import ZipTrace
+from pydantic import ValidationError
+from services.document_service import DocumentService
 from services.dynamo_service import DynamoDBService
 from services.s3_service import S3Service
 from utils.audit_logging_setup import LoggingService
-from utils.exceptions import ManifestDownloadException
+from utils.exceptions import (DocumentManifestServiceException,
+                              DynamoDbException)
 
 logger = LoggingService(__name__)
 
 
 class DocumentManifestService:
-    def __init__(
-        self,
-        nhs_number: str,
-        documents: list[DocumentReference],
-        zip_output_bucket: str,
-        zip_trace_table: str,
-    ):
+    def __init__(self, nhs_number):
         self.s3_service = S3Service()
         self.dynamo_service = DynamoDBService()
+        self.document_service = DocumentService()
+
         self.nhs_number = nhs_number
-        self.documents = documents
-        self.zip_output_bucket = zip_output_bucket
-        self.zip_trace_table = zip_trace_table
+        self.documents: list[DocumentReference] = []
```

Comment:
I don't feel this should be object level

Reply:
It's shared across multiple class methods so the alternative would be to pass it in as a parameter to said class methods, with that being the case keeping it at object level so it's methods have access to it seems sensible

Reply:
I agree, this one can be removed as it's only really used in one function